### PR TITLE
Sidebar validation next to fields

### PR DESCRIPTION
**Pull request by piekczyk**
# [Sidebar validation next to fields](https://app.shortcut.com/oazo-apps/story/4438/display-validation-messages-next-to-relevant-field)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- errors / warnings related exactly to specific field are now displayed next to it
  
## How to test 🧪
  <Please explain how to test your changes>
- test all possible flows and try to trigger all possible warnings / errors

**Use NewComponents feature toggle flag**

### CODE REVIEW

File: components/vault/VaultErrors.tsx [hunk 1]
```diff
@@ -79,12 +80,12 @@ export function VaultErrors({
...
         return translate('withdraw-amount-exceeds-free-collateral', {
           maxWithdrawAmount: formatCryptoBalance(maxWithdrawAmount),
-          token: vault?.token,
+          token: vault?.token || token,
         })
       case 'withdrawAmountExceedsFreeCollateralAtNextPrice':
         return translate('withdraw-amount-exceeds-free-collateral-at-next-price', {
           maxWithdrawAmount: formatCryptoBalance(maxWithdrawAmount),
-          token: vault?.token,
+          token: vault?.token || token,
```
Comment (johnnieskywalker) [inline components/vault/VaultErrors.tsx:88]: What is the purpose of `vault.token || token` ? Can't it be just obtained from ilkData and passed as `token` ?

**Comment by johnnieskywalker** [comment]
Managing vault scenarios

image

image

image

File: components/vault/VaultErrors.tsx (hunk 1 above)
Comment (piekczyk) [reply components/vault/VaultErrors.tsx]: very good point, let me fix that

**Comment by johnnieskywalker** [comment]
Opening vault scenarios:
image

image
image

image

image

**Review by johnnieskywalker — APPROVED** [review]
:seal: of approval

**Review by piotrkonowrocki — DISMISSED** [review]
I really like the idea an execution to keep components with errors and warning in the same place as input fields and also excluding errors list to an array.

I'm giving my approval and but I have only one comment and you can do with it whatever you want 😉 All methods that were extracting data from big object to smaller one to be used as props/params in sidebars were named `extractSomething`, not `pickSomething`. I think it would be good for the sake of coherency to rename them.

**Review by piotrkonowrocki — APPROVED** [review]

**Pull request merged**